### PR TITLE
Add gitcliff and reuse

### PR DESCRIPTION
**Pull request by Al-an-21**

### CODE REVIEW

**Comment by macpijan** [comment]
@Al-an-21 Please take a look at your MR before submitting for review. It contains much more commmits than just your changes, and in general cannot be reviewed in a sane way. It is likely that you wanted to target `develop` branch - adjusted that.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+<!--
+SPDX-FileCopyrightText: 2024 3mdeb <contact@3mdeb.com>
+
+SPDX-License-Identifier: MIT
```
Comment (macpijan) [inline README.md:4]: Where does the MIT came from @Al-an-21 ? We clearly already ship Apache license in this repo: https://github.com/Dasharo/open-source-firmware-validation/blob/develop/LICENSE

**Review by macpijan — CHANGES_REQUESTED** [review]
I have a really difficult time finding git cliff changes here. If there are any, please split them into individual commits from `reuse` changes, so it can be spotted and reviewed separately from auto changes to SPDX headers.

On top of that, if there are some I would appreciate:
- some comments explaining which `ruese` commands have been used to modify the source tree
- some comments explaining whether we needed to do some changes manually, on top of the `reuse` changes

**Comment by Al-an-21** [comment]
I’ve organized the files and split them across two commits for clarity. Additionally, I updated the license from MIT to Apache 2.0 and expanded the README.md with detailed instructions on using reuse commands for license compliance.

**Comment by macpijan** [comment]
> I’ve organized the files and split them across two commits for clarity. Additionally, I updated the license from MIT to Apache 2.0 and expanded the README.md with detailed instructions on using reuse commands for license compliance.

I fail to notice that.

reuse and cliff tools are still mixed up in the same commits, which was my main concern

image

**Comment by macpijan** [comment]
@Al-an-21 A friendly reminder that it has not been fix still :disappointed:

**Comment by DaniilKl** [comment]
Continuation here https://github.com/Dasharo/open-source-firmware-validation/pull/633.

**Pull request closed without merging**